src/Conversion/ONNXToKrnl/ONNXToKrnlCommon.hpp

```diff
@@ -72,12 +72,13 @@ struct OnnxToKrnlBuilder : public OnnxBuilder {
 // Common functions used when lowering the ONNX frontend dialect to KRNL.
 //===----------------------------------------------------------------------===//
 
-/// Check is all dimensions are known at compile time.
-bool hasAllConstantDimensions(MemRefType type);
-
-/// Check is all operands are scalar values at compile time.
+/// Check if all operands are scalar values at compile time.
 bool hasAllScalarValues(ArrayRef<Value> values);
 
+/// Check if the value is a KrnlGlobalOp with a dense attribute of positive
+/// constants.
```

Comment:
It seems value 0 is OK, so `positive constants` => `non-negative constants`.

Reply:
Renamed function to `indicesAreNonNegative` and adjusted comment.